retention messaging (#422)
adaptyteam/adapty-docs: src/content/docs/version-3.0/retention-messaging-create.mdx

---
title: "Create a retention message"
description: "Author a default retention message in Adapty: name it, pick products, write the title and body, add an optional image, localize it, and submit it for Apple's review."
metadataTitle: "Create a retention message | Adapty Docs"
---

:::link
Main article: [Retention Messaging](retention-messaging.md)
:::

> A retention message is what a subscriber sees on the iOS Cancel Subscription screen. You create a default message — a title and body, optionally with an image. Adapty shows it per product and locale when a subscriber cancels.

Reviewer: ```suggestion
A retention message is what a subscriber sees on the iOS Cancel Subscription screen. You create a default message — a title and body, optionally with an image. When a subscriber cancels, Apple shows the message that matches their product and locale.
```

> A retention message is what a subscriber sees on the iOS Cancel Subscription screen. You create a default message — a title and body, optionally with an image. Adapty shows it per product and locale when a subscriber cancels.

Reviewer: Ещё вопрос — мы как будто не объясняем, относительно чего дефолт? А что по недефолтным?

Reviewer: ещё я бы пояснила, что the default is the required fallback that must exist before promo/plan-switch messages

Reviewer: ещё раз перечитала — мы пишем сейчас так, будто только дефолтные сообщения и бывают. надо бы сначала объяснить в целом логику дефолтных vs остальных, а потом в целом писать, как создать любое сообщение

## Before you begin

- Create at least one [product](create-product) for the message to cover.
- [Set up Retention Messaging](retention-messaging-setup) and test it in sandbox environment.

## Create a default message

> To start, in the Adapty Dashboard go to **Retention Messaging**, then click **Create Default**.

Reviewer: ```suggestion
In the Adapty Dashboard, go to **Retention Messaging**, then click **Create Default**.
```

{/* screenshot here */}

> Choose where the message runs:

Reviewer: ```suggestion
Choose the environment for the message:
```

- **Create a sandbox message**: For testing. Sandbox messages never reach real users.
- **Create a production message**: For real users. Available once you've set your production URL and passed the sandbox performance test.

The message form opens with three steps: **General**, **Content**, and **Localization**.

### General

- **Name**: An internal name for the message, up to 60 characters.
- **Products**: The products this message covers. Add up to 20.

Click **Next**.

{/* screenshot here */}

### Content

Write the message in English. You can localize it in the next step.

- **Title**: Up to 66 characters.
- **Body**: Up to 144 characters.

> - **Image**: Optional. A PNG with the maximum width of 3840px and 160–2160px height, no transparency. Add **Alt text** to describe it.

Reviewer: не совсем
exactly 3840px wide, 160–2160px tall, no transparency, up to 10MB.

Click **Next**.

> {/* screenshot here */}

Reviewer: можно ещё про превью кратко написать тут

### Localization

> Adapty translates the message into your other locales.

Reviewer: ```suggestion
Use AI translation to fill in your other locales:
```

1. Click **Add Locale**, then choose the locales to add.
2. Click **AI Translate All Cells** to fill the translations.
3. Click any cell to edit a translation manually.

{/* screenshot here */}

## Submit for review

> When the message is ready, click **Submit To Review**. Apple reviews it, and its state changes to **In review**, then **Live** or **Rejected**. Production stays locked until Apple approves at least one locale.

Reviewer: ```suggestion
When the message is ready, click **Submit To Review**. Apple reviews the message. Its state changes to **In review**, then to **Live** or **Rejected**. The message doesn't go live for production users until Apple approves at least one locale.
```

кстати, а что будет, если у нас какие-то локали не аппрувнуты? фолбэк на английский?
